### PR TITLE
py-pybind11: add v2.13.5

### PR DESCRIPTION
**Pull request by adamjstewart**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pybind11/package.py [hunk 1]
```diff
@@ -23,10 +23,17 @@ class PyPybind11(CMakePackage, PythonExtension):
     homepage = "https://pybind11.readthedocs.io"
     url = "https://github.com/pybind/pybind11/archive/refs/tags/v2.10.1.tar.gz"
     git = "https://github.com/pybind/pybind11.git"
+    tags = ["build-tools"]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybind11/package.py]: Possibly controversial. I want to build my DAG with numpy 2. However, contourpy requires a brand new version of pybind11 to support numpy 2, whereas PyTorch still requires an older version. So we either:

1. allow multiple versions of pybind11 in the DAG when it is a build-only dependency, or
2. unpin the version of pybind11 used in the PyTorch build

**Review by tldahlgren — DISMISSED** [review]
Confirmed the 6 new version sha256s.

**Comment by wdconinc** [comment]
See also https://github.com/spack/spack/pull/45872.

File: var/spack/repos/builtin/packages/py-pybind11/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pybind11/package.py]: Attempting 2 in #46266

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pybind11/package.py]: 2 seems to work, this might not be necessary.

**Comment by adamjstewart** [comment]
Is this ready to merge?

**Review by wdconinc — APPROVED** [review]
Previously approved, and the only potentially controversial change was removed again, so approving again.

**Pull request merged**